### PR TITLE
cache-hash: add test_filter and test_name_prefix

### PR DESCRIPTION
**Pull request by travisstaloch**
Before this commit `zig test file.zig` followed by `zig test file.zig --test-filter foo` wasn't being respected and the tests weren't being rebuilt.

It looks like this isn't always working. Further testing seems like something is still missing.  I would expect the following to fail on `./zig test test.zig --test-filter fail`:

```zig
// test.zig
const std = @import("std");
test "pass" {
    std.testing.expect(true);
}
test "fail" {
    std.testing.expect(false);
}
```

```sh
$ ./zig test test.zig --test-filter pass
All 1 tests passed.
...
$ ./zig test test.zig --test-filter fail
All 1 tests passed.

$ rm -rf zig-cache

$ ./zig test test.zig --test-filter fail
Test [1/1] test "fail"... test failure
```

### CODE REVIEW

**Comment by travisstaloch** [comment]
Now this seems to be working as expected. I should have testing previously before pushing. :blush: 
```sh
$ ./zig test test.zig --test-filter pass
All 1 tests passed.

$ ./zig test test.zig --test-filter fail
Test [1/1] test "fail"... test failure
```

File: src/Compilation.zig [hunk 1]
```diff
@@ -2590,6 +2590,8 @@ fn updateStage1Module(comp: *Compilation, main_progress_node: *std.Progress.Node
     man.hash.add(comp.emit_llvm_ir != null);
     man.hash.add(comp.emit_analysis != null);
     man.hash.add(comp.emit_docs != null);
+    if (comp.test_filter != null) man.hash.add(comp.test_filter.?);
+    if (comp.test_name_prefix != null) man.hash.add(comp.test_name_prefix.?);
```
Comment (andrewrk) [inline src/Compilation.zig:2594]:
```suggestion
    man.hash.addOptionalBytes(comp.test_filter);
    man.hash.addOptionalBytes(comp.test_name_prefix);
```

**Review by andrewrk — CHANGES_REQUESTED** [review]
Thanks for noticing the problem, and for the patch! There's a utility function for that type - no need to modify the `add` function :)

File: src/Compilation.zig (hunk 1 above)
Comment (travisstaloch) [reply src/Compilation.zig:2594]: Of course. Should have spotted that with zls. Thanks. Updated.

**Pull request merged**